### PR TITLE
mark dataset test static for test index

### PR DESCRIPTION
**Pull request by LHT129**
- fix some typo error
- clean code
- mark dataset static for test index to speed up functest

### CODE REVIEW

File: tests/test_index.h [hunk 1]
```diff
@@ -170,51 +174,51 @@ class TestIndex {
...
-        indexs[key] = {index, status};
+    SaveIndex(const std::string& key, const IndexPtr& index, IndexStatus status) const {
+        indexes[key] = {index, status};
     }
 
     mutable int dataset_base_count{1000};
 
 private:
-    mutable std::unordered_map<std::string, std::shared_ptr<TestDataset>> datasets;
+    static std::unordered_map<std::string, TestDatasetPtr> datasets;
```
Comment (inabao) [inline tests/test_index.h:219]: will there be concurrency issues here?

File: tests/test_index.cpp [hunk 2]
```diff
@@ -32,6 +32,8 @@ Intersection(const int64_t* x, int64_t x_count, const int64_t* y, int64_t y_coun
     return result;
 }
 
+std::unordered_map<std::string, TestDatasetPtr> TestIndex::datasets = {};
```
Comment (inabao) [inline tests/test_index.cpp:35]: why do we need to make it static?

Comment (LHT129) [reply tests/test_index.cpp:35]: dataset may be same for all test

File: tests/test_index.h (hunk 1 above)
Comment (LHT129) [reply tests/test_index.h:219]: no concurrent test for this now

**Review by wxyucs — APPROVED** [review]
lgtm

File: tests/test_index.cpp [hunk 3]
```diff
@@ -151,23 +151,23 @@ TestIndex::TestContinueAdd(IndexPtr index, int64_t dim, int64_t count, bool expe
...
+TestIndex::TestKnnSearch(const IndexPtr& index,
+                         const TestDatasetPtr& dataset,
                          const std::string& search_param,
                          int topk,
                          float recall,
                          bool expected_success) {
-    auto querys = dataset->query_;
-    auto query_count = querys->GetNumElements();
-    auto dim = querys->GetDim();
+    auto queries = dataset->query_;
```
Comment (jiaweizone) [inline tests/test_index.cpp:160]: we need rename dataset `query_` to `query` ?

**Pull request merged**